CMakePresets.json

```diff
@@ -0,0 +1,172 @@
+{
+	"version": 3,
+	"cmakeMinimumRequired": {
+		"major": 3,
+		"minor": 19,
+		"patch": 0
+	},
+	"configurePresets": [
+		{
+			"name": "debug-flags",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_BUILD_TYPE": "Debug",
+				"CMAKE_C_FLAGS": "-fno-omit-frame-pointer -fno-optimize-sibling-calls -gfull -O0",
+				"CMAKE_CXX_FLAGS": "-fno-omit-frame-pointer -fno-optimize-sibling-calls -gfull -O0"
+			},
+      "environment": {
+        "VCPKG_SUFFIX": ""
+      }
+		},
+		{
+			"name": "release-flags",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_BUILD_TYPE": "Release"
+			},
+      "environment": {
+        "VCPKG_SUFFIX": "-rel"
+      }
+		},
+    {
+      "name": "arm64",
+      "hidden": true,
+      "environment": {
+        "VCPKG_ARCH": "arm64"
+      },
+      "architecture": {
+        "value": "arm64",
+        "strategy": "external"
+      }
+    },
+    {
+      "name": "x86_64",
+      "hidden": true,
+      "environment": {
+        "VCPKG_ARCH": "x64"
+      },
+      "architecture": {
+        "value": "x64",
+        "strategy": "external"
+      }
+    },
+		{
+			"name": "linux",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_C_COMPILER": "$env{VCPKG_ROOT}/installed/x64-linux$env{VCPKG_SUFFIX}/tools/llvm/clang",
+				"CMAKE_CXX_COMPILER": "$env{VCPKG_ROOT}/installed/x64-linux$env{VCPKG_SUFFIX}/tools/llvm/clang++"
+			},
+      "environment": {
+        "VCPKG_TARGET": "$env{VCPKG_ARCH}-linux"
+      },
+			"condition": {
+				"type": "equals",
+				"lhs": "Linux",
+				"rhs": "${hostSystemName}"
+			}
+		},
+		{
+			"name": "macos",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_C_COMPILER": "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang",
+				"CMAKE_CXX_COMPILER": "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"
```

Comment:
Is this really necessary?

Shouldn't we just let CMake pick the compiler on its own?

I think the only issue we have is with GCC, but we should mention that in the README.

Reply:
@ekilmer I would prefer to be able to specify something like `which clang` and `which clang++`, if possible, due to asan working with that on macOS, but not with vcpkg clang on macOS.

Reply:
That should go in the `CMakeUserPresets.json` because your clang could have a different path than someone else.

Reply:
Artem copied this absolute path from me because I was trying to diagnose a gRPC issue. Normally I do `which clang` and `which clang++`; another alternative for unix systems would be `which cc` and `which c++`. I guess if default CMake on macOS with vcpkg finds the system compiler then I'd be happy.

Reply:
I have not tested a MacOS build but would be ok eliminating this; I was worried about it finding the "correct" clang to build & link; although I guess it could be fine to leave this out.

Reply:
How about `/usr/bin/cc` and `/usr/bin/c++`?

Reply:
Also -- this just sets the default; it should still be possible to override it via `-DCMAKE_C_COMPILER=...`... need to see if just setting CC and CXX would also override it.